sdk/appconfiguration/app-configuration/src/featureFlag.ts

```diff
@@ -0,0 +1,345 @@
+ */
+function isFeatureFlagPercentageClientFilter(
+  clientFilter: any
+): clientFilter is FeatureFlagPercentageClientFilter {
+  return clientFilter.name === "Microsoft.Percentage";
+}
+
+/**
+ * Type of FeatureFlag based on the client filters.
+ */
+export type FeatureFlagType<
+  T extends "targeting" | "timeWindow" | "percentage"
+> = T extends "targeting"
+  ? FeatureFlagTargetingClientFilter
+  : T extends "timeWindow"
+  ? FeatureFlagTimeWindowClientFilter
+  : T extends "percentage"
+  ? FeatureFlagPercentageClientFilter
+  : never;
+
+/**
+ * This helper method tells you if the given client filter is a FeatureFlagClientFilter.
+ */
+export function isFeatureFlagClientFilter<T extends "targeting" | "timeWindow" | "percentage">(
+  type: T,
+  obj: any
+): obj is FeatureFlagType<T> {
+  switch (type) {
+    case "targeting":
+      return isFeatureFlagTargetingClientFilter(obj);
+    case "timeWindow":
+      return isFeatureFlagTimeWindowClientFilter(obj);
+    case "percentage":
+      return isFeatureFlagPercentageClientFilter(obj);
+    default:
+      throw new Error(`Invalid feature flag filter type ${type}`);
+  }
+}
+
+export function deserializeFeatureFlag(setting: ConfigurationSetting): FeatureFlag | undefined {
+  if (!isFeatureFlag(setting) || !setting.value) {
+    return undefined;
+  }
+
+  let jsonFeatureFlag: JsonFeatureFlag;
+
+  try {
+    jsonFeatureFlag = JSON.parse(setting.value) as JsonFeatureFlag;
+  } catch (err) {
+    // best effort - if it doesn't deserialize properly we'll just let it "degrade"
+    // to be treated as a ConfigurationSetting.
+    return undefined;
+  }
+
+  // update (in-place) the feature flag specific properties
+  const featureFlag = setting as Omit<FeatureFlag, keyof ConfigurationSetting>;
+  featureFlag.conditions = convertJsonConditions(jsonFeatureFlag.conditions);
+  featureFlag.enabled = jsonFeatureFlag.enabled;
+  featureFlag.description = jsonFeatureFlag.description;
+
+  return setting;
+}
+
+export function serializeFeatureFlagParam(setting: FeatureFlagParam): ConfigurationSettingParam {
+  const value: JsonFeatureFlag & { id: string } = {
+    id: setting.key.replace(featureFlagPrefix, ""),
+    description: setting.description,
+    enabled: setting.enabled,
+    conditions: convertToJsonConditions(setting.conditions)
+  };
+  const configurationSetting: ConfigurationSettingParam = {
+    key: setting.key,
+    label: setting.label,
+    contentType: setting.contentType,
+    etag: setting.etag,
+    tags: setting.tags,
+    value: JSON.stringify(value)
+  };
+  return configurationSetting;
+}
+
+/**
+ * @internal
+ */
+export function convertJsonConditions(
+  conditions: JsonFeatureFlag["conditions"]
+): FeatureFlag["conditions"] {
+  const clientFilters = !conditions.client_filters
+    ? []
+    : conditions.client_filters.map((jsonFilter) => {
+        if (isJsonFeatureFlagTargetingClientFilter(jsonFilter)) {
+          // try not to slice any unknown attributes
```

Comment:
I'm assuming this is more of a TODO - I don't see any "slicing prevention" code here.

Reply:
I've confirmed there are no unknown attributes.
But I left the comment anyways just to allow if we ever had to support such in the future.

Reply:
Maybe it's better to remove the comment instead.